internal/services/mssql/mssql_virtual_machine_resource.go

```diff
@@ -606,7 +606,9 @@ func resourceMsSqlVirtualMachineRead(d *pluginsdk.ResourceData, meta interface{}
 	if model := resp.Model; model != nil {
 		if props := model.Properties; props != nil {
 			d.Set("virtual_machine_id", props.VirtualMachineResourceId)
-			d.Set("sql_license_type", string(*props.SqlServerLicenseType))
+			if licenceType := props.SqlServerLicenseType; licenceType != nil {
+				d.Set("sql_license_type", string(*licenceType))
+			}
```

Comment:
this field always needs to be set into the state:

```suggestion
			sqlLicenseType := ""
			if licenceType := props.SqlServerLicenseType; licenceType != nil {
				sqlLicenseType = string(*licenceType)
			}
			d.Set("sql_license_type", sqlLicenseType)
```